src/connection/database/actions.js

```diff
@@ -70,9 +70,11 @@ export function signUp(id) {
       additionalSignUpFields(m).forEach(x => {
         if (x.get('type') !== 'textinfo') {
           if (x.get('isExtra')) {
-            params.user_metadata.extraFields[x.get('name')] = c.getFieldValue(m, x.get('name'));
+            params.user_metadata.extraFields[x.get('name')] = c
```

Comment:
is 'name' the combination of first name and last name? do we need to trim them separately?

Reply:
Name is the name of the parameter, so they get trimmed separately. 

I think they use immutable.js in some places so I think x is an immutable object
 meaning x.get('name') is to be understood as x.name.

Reply:
```
additionalSignUpFields = Immutable.fromJS(additionalSignUpFields).map(x =>
      x.filter(y => y !== undefined)
    );
```
It's an immutable map